client/src/main/java/io/cloudsoft/winrm4j/client/WinRmClientBuilder.java

```diff
@@ -14,6 +15,15 @@
 
 public class WinRmClientBuilder {
     private static final java.util.Locale DEFAULT_LOCALE = java.util.Locale.US;
+    /**
+     * Timeout applied by default on client side for the opening of the socket (0 meaning infinite waiting).
+     */
+    private static final Long DEFAULT_CONNECTION_TIMEOUT = 0l;
```

Comment:
Looking at the existing default behaviour in `org.apache.cxf.transports.http.configuration.HttpClientProxy` (I checked in all of v3.2.7, v3.2.8 and v3.3.2), the `connectionTimeout` defaults to `null`. If not set, or set to null, then `HttpClientProxy.getConnectionTimeout()` returns `30000L` (rather than `0L`).

Is there a reason to change that default for existing users of `winmj4`? If not, I suggest setting the default to `null` here, as you do with `DEFAULT_RECEIVE_TIMEOUT`, and thus removing `Objects.requireNonNull` in the builder. Or changing this default to `30000L` to match the existing default behaviour.